### PR TITLE
Merge locus changes

### PR DESCRIPTION
**Pull request by ayrton04**
Rebased the changes from upstream/indigo-devel -> origin/indigo-devel, and am now applying them to kinetic-devel. This will let us move to kinetic and retain our changes.

@locusrobotics/robot-software-team

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp [hunk 1]
```diff
@@ -232,17 +301,37 @@ namespace gazebo
...
     linear.x = (pose.pos.x - last_odom_pose_.pos.x) / step_time;
     linear.y = (pose.pos.y - last_odom_pose_.pos.y) / step_time;
+
+    if (static_x_variance_)
+    {
+      odom_.pose.covariance[0] = x_variance_;
+    }
+    else
+    {
+      odom_.pose.covariance[0] += calculateScaledVelocityVariance(linear.x,
```
Comment (tappan-at-git) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:319]: Is there anything to stop these from growing without bounds?

**Comment by tappan-at-git** [comment]
LGTM. Should we set our simulation to use static variances?

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp (hunk 1 above)
Comment (ayrton04) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:319]: No, but that's realistic. The robot's raw odometry covariance _should_ be unbounded.

**Comment by ayrton04** [comment]
Nope. :)

File: gazebo_plugins/src/gazebo_ros_planar_move.cpp (hunk 1 above)
Comment (tappan-at-git) [inline gazebo_plugins/src/gazebo_ros_planar_move.cpp:319]: Freaky.

**Comment by bradleypowers** [comment]
LGTM.

**Pull request merged**